src/decn/decn.c

```diff
@@ -664,8 +673,8 @@ void add_decn(void){
 	}
 	//may need to rescale number
 	if (carry > 0){
-		assert(carry == 1);
 		exp_t curr_exp = get_exponent(&AccDecn);
+		assert(carry == 1);
 		rel = (AccDecn.exponent < 0); //is_neg?
```

Comment:
Any reason for changing the order here?

Reply:
SDCC 3.5.0 doesn't support C11 I guess. A statement (assert) after the variable declaration has AFAIK always been invalid C.

Reply:
Right, didn't notice the mixed code/declaration here.